contrib/add-capsule-header.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import sys
+import uuid
+import argparse
+import struct
+
+CAPSULE_FLAGS_PERSIST_ACROSS_RESET = 0x00010000
+CAPSULE_FLAGS_POPULATE_SYSTEM_TABLE = 0x00020000
+CAPSULE_FLAGS_INITIATE_RESET = 0x00040000
+
+def main(args):
+
+    # parse GUID from command line
+    try:
+        guid = uuid.UUID(args.guid)
+    except ValueError as e:
+        print(e)
+        return 1
+    try:
+        with open(args.bin, 'rb') as f:
+            bin_data = f.read()
+    except FileNotFoundError as e:
+        print(e)
+        return 1
+
+    # check if already has header
+    hdrsz = struct.calcsize('<16sIII')
+    if len(bin_data) >= hdrsz:
+        hdr = struct.unpack('<16sIII', bin_data[:hdrsz])
+        imgsz = hdr[3]
+        if imgsz == len(bin_data):
+            print('Replacing existing CAPSULE_HEADER of:')
+            guid_mixed = uuid.UUID(bytes_le=hdr[0])
+            hdrsz_old = hdr[1]
+            flags = hdr[2]
+            print('GUID:      %s' % guid_mixed)
+            print('HdrSz:     0x%04x' % hdrsz_old)
+            print('Flags:     0x%04x' % flags)
+            print('PayloadSz: 0x%04x' % imgsz)
+            bin_data = bin_data[hdrsz_old:]
+
+    # set header flags
+    flags = 0
```

Comment:
I think you want to set default flags using your defines at the top (and show a message that you're setting them).  This is for use with fwupd and I don't see a case that we would ever want absolutely _no_ flags.

Reply:
For fwupd, we fixup the firmware with the ESRT flags -- but what is the right thing to do here by default? Is `CAPSULE_FLAGS_PERSIST_ACROSS_RESET | CAPSULE_FLAGS_INITIATE_RESET` sane?

Reply:
Yes I think so.